source/MaterialXCore/Geom.h

```diff
@@ -342,6 +348,97 @@ class GeomAttr : public ValueElement
     static const string CATEGORY;
 };
 
+/// @class GeomProp
+/// An element representing a declaration of geometric input data.
+/// Can be used as a child to Input elements to declare a geometric node
+/// to be used if no explicit connection is set.
+/// Can also be used as children to NodeDef elements to declare internal
```

Comment:
I assume the implicit assumption is that geomprop can never be declared without a parent, and that type is specified by the parent. 

Is there an example of how this can be a child of a nodedef. If they are direct children then where is the type specifeid?

Reply:
I see the nodedef reference is for replacement of internalgeomprops as in your example:
```
<nodedef name="ND_someothernode" node="someothernode" type="color3">
    <input name="color" type="color3" value="0.18, 0.18, 0.18"/>
    <geomprop name="normal" space="world"/>
    <geomprop name="tangent" space="world" index="1"/>
    <geomprop name="texcoord" index="1"/>
</nodedef>
```
So the question about how to determine type applies to this property that is being replaced.

Reply:
My thinking was that type is derived from the referenced geometric node, for example "normal" being of type vector3. But I think you are right, in some cases this is not enough. For example the texcoord node can be both vetor2 and vector3. So I should make TypedElement be the parent class instead, to add type information. Thanks

Reply:
Thanks. I assume it's an error then if the type of the geomprop (if type is explicitly specified) does not match the type of an input.